bfsscp/bfsscp.go

```diff
@@ -43,26 +43,22 @@ import (
 )
 
 func init() {
-	bfs.Register("ssh", register)
-	bfs.Register("scp", register)
-}
-
-// register allows for registering more schemes with SCP
-func register(_ context.Context, u *url.URL) (bfs.Bucket, error) {
-	query := u.Query()
-	address := net.JoinHostPort(u.Host, u.Port())
-
-	username, password := "", ""
-	if u.User != nil {
-		username = u.User.Username()
-		password, _ = u.User.Password()
-	}
+	bfs.Register("scp", func(_ context.Context, u *url.URL) (bfs.Bucket, error) {
+		query := u.Query()
+		address := net.JoinHostPort(u.Hostname(), u.Port())
```

Comment:
I may not have the full context, but what happened with just `u.Host`?

https://golang.org/pkg/net/url/#URL

```go
type URL struct {
    ...
    Host       string    // host or host:port
    ...
}
```

Reply:
> or host:port
Is the issue. we were always generating `host:port:port`

Reply:
@ajn that's because you were doing `net.JoinHostPort(u.Host, u.Port())` previously.

You could drop the `net.JoinHostPort` and just use `u.Host` directly.